woocommerce/woocommerce-ios
# REST API: 2FA UI and WPCom login integration

Part of #8918 


## Description

This PR continues the custom WPCom login flow for the Jetpack setup feature:
- New UI for the 2FA login screen.
- Integrated `LoginFacade` from `WordPressAuthenticator` to handle login with email, password, and 2FA code.
- Handled requesting SMS OTP code.
- Also: updated `AccountCreationFormFieldView` to accept the header label to be nil.

Flow reference: [pe5sF9-1er-p2]
## Testing instructions

Prerequisite: Make sure that you have access to a site with WooCommerce and no Jetpack.
- Log out of the app or skip onboarding if needed.
- Tap "Enter your site address" on the prologue screen and proceed with the address of your test site.
- Log in with the credentials of an admin.
- When the login completes, tap the Jetback benefit banner at the bottom of the Dashboard screen.
- Tap Log In on the benefit modal.
- Enter the email address of a WPCom account with a password and no 2FA setup yet on the presented WPCom email login screen.
- Tap Install/Connect Jetpack. Enter an incorrect password and notice that an error alert is displayed for incorrect credentials.
- Enter the correct password, notice in Xcode console: `✅ Ready for Jetpack setup`.
- Tap back and try again with the email address of a WPCom account with password and 2FA setup.
- Enter the correct password, notice that a new screen is displayed for the 2FA code.
- Enter an incorrect code, and an error alert should be displayed.
- Enter the correct code, notice in Xcode console: `✅ Ready for Jetpack setup`.

## Screenshots

<img src="https://user-images.githubusercontent.com/5533851/222676360-efa2e269-837a-4f42-b35a-e0ac8c5137c0.png" width=320 />


---
